sdk/keyvault/Azure.Management.KeyVault/tests/VaultOperationsTestsBase.cs

```diff
@@ -0,0 +1,194 @@
+        public static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
+
+        public string tenantId { get; set; }
+        public string objectId { get; set; }
+        public string applicationId { get; set; }
+        public string location { get; set; }
+        public string subscriptionId { get; set; }
+
+        public AccessPolicyEntry accPol { get; internal set; }
+        public string objectIdGuid { get; internal set; }
+        public string rgName { get; internal set; }
+        public Dictionary<string, string> tags { get; internal set; }
+        public Guid tenantIdGuid { get; internal set; }
+        public string vaultName { get; internal set; }
+        public VaultProperties vaultProperties { get; internal set; }
+
+
+        public VaultsClient VaultsClient { get; set; }
+        public ResourcesClient ResourcesClient { get; set; }
+        public ResourceGroupsClient ResourceGroupsClient { get; set; }
+        public UsersClient GraphUsersClient { get; set; }
+        public ProvidersClient ResourceProvidersClient { get; set; }
+
+
+        protected VaultOperationsTestsBase(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        protected async Task Initialize()
+        {
+            if (Mode == RecordedTestMode.Playback && Recording.IsTrack1SessionRecord())
+            {
+                this.tenantId = TestEnvironment.TenantIdTrack1;
+                this.subscriptionId = TestEnvironment.SubscriptionIdTrack1;
+                this.applicationId = TestEnvironment.ApplicationIdTrack1;
+            }
+            else
+            {
+                this.tenantId = TestEnvironment.TenantId;
+                this.subscriptionId = TestEnvironment.SubscriptionId;
+                this.applicationId = TestEnvironment.ClientId;
+            }
+
+            VaultsClient = GetVaultsClient();
+            ResourcesClient = GetResourcesClient();
+            ResourceGroupsClient = GetResourceGroupsClient();
+            ResourceProvidersClient = GetResourceProvidersClient();
+
+            if (Mode == RecordedTestMode.Playback)
+            {
+                this.objectId = Recording.GetVariable(ObjectIdKey, string.Empty);
+            }
+            else if (Mode == RecordedTestMode.Record)
+            {
+                var spClient = new Graph.Rbac.ServicePrincipalsClient(this.tenantId, TestEnvironment.Credential);
+                var servicePrincipalList = spClient.ListAsync($"appId eq '{this.applicationId}'");
+                await foreach (var servicePrincipal in servicePrincipalList)
+                {
+                    this.objectId = servicePrincipal.ObjectId;
+                    Recording.GetVariable(ObjectIdKey, this.objectId);
+                    break;
+                }
+            }
+            var provider = (await ResourceProvidersClient.GetAsync("Microsoft.KeyVault")).Value;
+            this.location = provider.ResourceTypes.Where(
```

Comment:
LOcation is available on TestEnvironment

Reply:
Prefer to set environment variables as less as possible for recording.

Reply:
You don't need to set them manually or at all, please use the New-TestResources script with `-OutFile` parameter.